src/ucp/core/ucp_request.inl

```diff
@@ -563,11 +563,11 @@ ucp_recv_desc_init(ucp_worker_h worker, void *data, size_t length,
     if (ucs_unlikely(am_flags & UCT_CB_PARAM_FLAG_DESC)) {
         /* slowpath */
         ucs_assert(priv_length <= UCP_WORKER_HEADROOM_PRIV_SIZE);
-        data_hdr           = UCS_PTR_BYTE_OFFSET(data, -data_offset);
-        rdesc              = (ucp_recv_desc_t *)data_hdr - 1;
-        rdesc->flags       = rdesc_flags | UCP_RECV_DESC_FLAG_UCT_DESC;
-        rdesc->priv_length = priv_length;
-        status             = UCS_INPROGRESS;
+        data_hdr               = UCS_PTR_BYTE_OFFSET(data, -data_offset);
+        rdesc                  = (ucp_recv_desc_t *)data_hdr - 1;
+        rdesc->flags           = rdesc_flags | UCP_RECV_DESC_FLAG_UCT_DESC;
+        rdesc->uct_desc_offset = UCP_WORKER_HEADROOM_PRIV_SIZE - priv_length;
```

Comment:
assert this is >= 0

Reply:
no need, there is an assert that UCP_WORKER_HEADROOM_PRIV_SIZE  > priv_length already

Reply:
ok